mlir/include/mlir/Dialect/MemRef/IR/MemRefOps.td

```diff
@@ -987,8 +987,6 @@ def MemRef_ExtractStridedMetadataOp : MemRef_Op<"extract_strided_metadata", [
 
     ::mlir::Value getViewSource() { return getSource(); }
   }];
-
-  let hasFolder = 1;
```

Comment:
Also add to the description of the op why the offsets and strides if constant shouldnt be folded away. Basically we want to make sure these dont get added back in by providing context as to why these shouldnt be added.